src/peerRatedPerformance.js

```diff
@@ -0,0 +1,239 @@
+
+    return (
+        <CustomField>
+            <Text>
+                <StatusLozenge text={output} appearance={getLozengeApperance(output)}></StatusLozenge>
+            </Text>
+        </CustomField>
+    );
+};
+
+const PeerAssessedPerformanceEdit = () => {
+    
+    const {platformContext: {issueKey}} = useProductContext();
+    const [assigneeId] = useState(getIssueAssigneeID(issueKey));
+    const {accountId} = useProductContext();
+    const [fieldData] = useState(getPeerAssessedPerformanceData(issueKey));
+
+    // If current user is the assignee, keep return data the same
+    if (accountId == assigneeId) {
+        const onSubmit = (formValue) => {
+            var accounts;
+            var ratings;
+
+            // if first time peer rating performance for this issue, set properties to empty array
+            if (fieldData === null || typeof fieldData === 'undefined') {
+                accounts = [];
+                ratings = [];
+            }
+            else {
+                accounts = fieldData.accountId;
+                ratings = fieldData.peerAssessedPerformanceRating;
+            };
+            return {
+                "accountId": accounts,
+                "peerAssessedPerformanceRating": ratings
+            };
+        };
+        
+        return (
+            <CustomFieldEdit onSubmit={onSubmit} header="How would you rate the performance of your peer?" width="medium" >
+                <Text>Cannot peer assess yourself!</Text>
+            </CustomFieldEdit>
+        );
+    };
+
+    // Else, let user rate performance like self assessment 
+    const onSubmit = (formValue) => {
+        
+        var accounts;
+        var ratings;
+
+        // if first time peer rating performance for this issue, set properties to empty array
+        if (fieldData === null || typeof fieldData === 'undefined') {
+            accounts = [];
+            ratings = [];
+        }
+        else {
+            accounts = fieldData.accountId;
+            ratings = fieldData.peerAssessedPerformanceRating;
+        };
+
+        // If user already rated performance, update thier previous rating
+        var exists = false;
+        for (let i = 0; i < accounts.length; i++) {
+            if (accounts[i] == accountId) {
+                ratings[i] = formValue.myPerformanceRating;
+                exists = true;
+            }
+        }
+
+        // If user has not rated performance before, add their rating
+        if (!exists) {
+            accounts.push(accountId);
+            ratings.push(formValue.myPerformanceRating);
+        }
+
+        return {
+            "accountId": accounts,
+            "peerAssessedPerformanceRating": ratings
+        };
+    }
+
+    return (
+        <CustomFieldEdit onSubmit={onSubmit} header="How would you rate the performance of your peer?" width="medium" >
+            <Select label="Select one of the options below to rate the performance of your peer." name="myPerformanceRating">
```

Comment:
is the `name` supposed to be `myPeerPerformanceRating`, still works though...

Reply:
Yeah it should have been, have renamed it now!